mavisp/core.py

```diff
@@ -129,9 +132,47 @@ def _dataset_table(self, include, exclude):
                         self.log.debug("There is no review status field in metadata file")
                         review_status = None
 
-                df_list.append((system, uniprot_ac, refseq_id, review_status, mode, mutation_list, curators))
-
-        main_df = pd.DataFrame.from_records(df_list, columns=['system', 'uniprot_ac', 'refseq_id','review_status', 'mode', 'mutations', 'curators'])
+                    try:
+                        # Check if the ensemble sources is a list, if it's not, it's equal to None
+                        if isinstance(metadata["ensemble_sources"],list):
+                            ensemble_sources = ", ".join([str(value) for value in metadata["ensemble_sources"]])
+                        else:
+                            ensemble_sources = None
+                            self.log.debug("Ensemble sources are in the wrong format")
+                    except KeyError:
+                        self.log.debug("There are no ensemble sources in metadata file")
+                        ensemble_sources = ""
+                    try:
+                        if isinstance(metadata["ensemble_size_foldx"],list):
+                            ensemble_size_foldx = ", ".join([str(value) for value in metadata["ensemble_size_foldx"]])
+                        else:
+                            ensemble_size_foldx = None
+                            self.log.debug("Ensemble size Foldx is in the wrong format")
+                    except KeyError:
+                        self.log.debug("There is no ensemble size for FoldX field in metadata file")
+                        ensemble_size_foldx = ""
+                    try:
+                        if isinstance(metadata["ensemble_size_rosetta"],list):
+                            ensemble_size_rosetta = ", ".join([str(value) for value in metadata["ensemble_size_rosetta"]])
+                        else:
+                            ensemble_size_rosetta = None
+                            self.log.debug("Ensemble size Rosetta is in the wrong format")
+                    except KeyError:
+                        self.log.debug("There is no ensemble size for Rosetta field in metadata file")
+                        ensemble_size_rosetta = ""
+                    try:
+                        if len(metadata["ensemble_size_foldx"]) == len(metadata["ensemble_size_rosetta"]) == len(metadata["ensemble_sources"]):
```

Comment:
I would invert this logic from

```
if ..
    do nothing
else ..
    do something
```

to

```
if not ...
    do something
```

so something like

```
if lenmetadata['a'] != lenmetadata['b'] or lenmetadata['a'] != lenmetadata['c'] or lenmetadata['b'] != lenmetadata['c']:
    a = None
    ...
```

or probably more succinctly:

```
if not len(set([ len(metadata[m]) for m in ["ensemble_size_foldx", "ensemble_size_rosetta", "ensemble_sources"])) == 1:
```

this is because of [De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws):

```
! (A and B) = !A or !B
```